### DISCUSSION TITLE
Feature to create one control per column in the Customize tab, which can be used for different configuration for Charts

### DISCUSSION THREAD

**Discussion by webobite** [apache/superset#24679] [Ideas]
There was a discussion over Slack for which I am facing a design blocker where we do not have much of a feature through which we can add one control per column being added in the tooltip. This would enable some customisation for charts in the UI

**1st Use Case**: Let's say, for the colour scheme, I want the individual colour to be added for each bar, in bar chart (this can be any other chart too) for that Superset doesn't provide much control, only one can do it add a colour theme at one place in the explorer tab

**2nd Use Case**: For each column plotted I want some customisation to be added in the tooltip, that control is also not available in Superset.

The **proposed idea to start exploring**:-  To add a feature shared control that could get attached to every column in the customize tab, the user and toggle on different control as and when needed.

I am yet to start with any implementation, Sharing here to start a conversation with the community.
I was hoping to get some more proposals/ideas to improve the same and get started with the same.

Thank You 
Subham

**Comment by webobite** [comment]
@rusackas  adding you to this discussion, as you were part of that conversation on slack.

**Comment by minhchau2236** [comment]
Hi @webobite , thanks for bringing this idea here, it's awesome! I have the same idea:

1. **Use case 1**: This should be very useful in some cases, each bar (line, pie, ,,,) has particular meaning, we need exact color for particular one. Currently we can go to edit Dashboard properties (label_colors) to achieve it. For normal user, it should be best UX to have UI control in the control panel, so that any user can have a clear color picker to pick color from color scheme or any color for series.

![image](https://github.com/apache/superset/assets/19927528/59a67b18-a43c-49da-bf2b-b9c55b91e754)
![image](https://github.com/apache/superset/assets/19927528/ef0b4f84-a5d8-4c94-954e-e37ac0f7c3c3)

2. **Use case 2**: This is also useful as well, we face some requirement to update the tooltip content, bar width, ... 

I am having some implementation on use case 1.

**Reply by webobite** [reply]
Yeah agree !

**Comment by minhchau2236** [comment]
Hi, here is a small demo for use case 1 that I have done so far:

https://github.com/apache/superset/assets/19927528/daefc2f2-4348-466b-9daf-70a5dcfdb036

From the above demo:
1. The Custom color set will override the color which is picked from the color scheme. Only for columns that have color defined, for others, they still keep the colors from color scheme. Don’t need to have every color picker for every column.
2. For the color picker, we will use the current built-in color picker from Superset, which has pre defined colors which are picked from particular color scheme. We also provide ability to pick ad-hoc color by color hex code as well.
